visualization/app/codeCharta/ui/codeMap/codeMap.arrow.service.ts

```diff
@@ -110,30 +108,38 @@ export class CodeMapArrowService
 		}
 	}
 
-	private showEdgesOfBuildings(edges: Edge[]) {
-		const node = this.threeSceneService.getHighlightedNode()
-		if (this.threeSceneService.getSelectedBuilding() && node) {
-			this.buildPairingEdges(this.threeSceneService.getSelectedBuilding().node, edges)
-			this.buildPairingEdges(node, edges)
-		} else if (node) {
-			this.buildPairingEdges(node, edges)
-		} else if (this.threeSceneService.getSelectedBuilding()) {
-			this.buildPairingEdges(this.threeSceneService.getSelectedBuilding().node, edges)
-		} else {
+	private isEdgeAppliableForBuilding(codeMapBuilding: CodeMapBuilding) {
+		return this.storeService.getState().dynamicSettings.edgeMetric !== "None" && !codeMapBuilding.node.flat
+	}
+
+	private showEdgesOfBuildings(edges: Edge[], hoveredbuilding?: CodeMapBuilding) {
```

Comment:
```suggestion
	private showEdgesOfBuildings(edges: Edge[], hoveredbuilding: CodeMapBuilding) {
```

The node should AFAIK always be defined?